src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -53,45 +53,17 @@ class FileService(
   }
 
   def getFileMetadata(consignmentId: UUID): Future[List[File]] = {
-    ffidMetadataRepository.getFFIDMetadata(consignmentId).map {
-      ffidMetadataAndMatchesRows =>
-        val ffidMetadataAndMatches: Map[FfidmetadataRow, Seq[FfidmetadatamatchesRow]] =
-          ffidMetadataAndMatchesRows.groupBy(_._1).view.mapValues(_.map(_._2)).toMap
-
-        fileMetadataRepository.getFileMetadata(consignmentId).map {
-          rows =>
-            rows.groupBy(_.fileid).map(entry => {
-              val propertyNameMap: Map[String, String] = entry._2.groupBy(_.propertyname)
-                .transform((_, value) => value.head.value)
-
-              val fileMetadataValues = FileMetadataValues(
-                propertyNameMap.get(SHA256ClientSideChecksum),
-                propertyNameMap.get(ClientSideOriginalFilepath),
-                propertyNameMap.get(ClientSideFileLastModifiedDate).map(d => Timestamp.valueOf(d).toLocalDateTime),
-                propertyNameMap.get(ClientSideFileSize).map(_.toLong),
-                propertyNameMap.get(RightsCopyright.name),
-                propertyNameMap.get(LegalStatus.name),
-                propertyNameMap.get(HeldBy.name),
-                propertyNameMap.get(Language.name),
-                propertyNameMap.get(FoiExemptionCode.name)
-              )
-              val fileId = entry._1
-              val fullFfidMetadata = ffidMetadataAndMatches.find(_._1.fileid == fileId).map {
-                case (metadata, matches) => FFIDMetadata(
-                  metadata.fileid,
-                  metadata.software,
-                  metadata.softwareversion,
-                  metadata.binarysignaturefileversion,
-                  metadata.containersignaturefileversion,
-                  metadata.method,
-                  matches.map(m => FFIDMetadataMatches(m.extension, m.identificationbasis, m.puid)).toList,
-                  metadata.datetime.getTime
-                )
-              }
-              File(fileId, fileMetadataValues, fullFfidMetadata)
-            }).toList
-        }
-    }.flatten
+    val fileMetadataService = new FileMetadataService(fileMetadataRepository, timeSource, uuidSource)
+    val ffidMetadataService = new FFIDMetadataService(ffidMetadataRepository, ffidMetadataMatchesRepository, timeSource, uuidSource)
```

Comment:
Could these be passed to this service's constructor, so that all the services are created in the same place?

Reply:
The only reason I didn't is because there's an existing method in `FileService` `addFile` which is directly using the `FileMetadataRepository` and I didn't want the service and the repository in the same constructor. What makes more sense now you say it is to add another method to `FileMetadataService` to add the static metadata and then all we need to have in the `FileService` constructor is the `FileMetadataService` and `FFIDMetadataService`
If this was a Java project I'd say this is facade territory but I think this is ok.